api-gateway/Dockerfile

```
@@ -0,0 +1,41 @@
+RUN sed -i 's/\r$//' gradlew && chmod +x gradlew
+
+RUN --mount=type=cache,target=/root/.gradle \
+    ./gradlew :api-gateway:dependencies --no-daemon
+
+COPY api-gateway/src api-gateway/src
+
+RUN --mount=type=cache,target=/root/.gradle \
+    ./gradlew :api-gateway:bootJar --no-daemon -x test
+
+FROM eclipse-temurin:21-jre-alpine
+RUN apk add --no-cache curl
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)